dev/com.ibm.ws.threading/src/com/ibm/ws/threading/internal/PolicyExecutorImpl.java

```diff
@@ -1176,6 +1177,71 @@ private void transferOrReleasePermit() {
+    @Override
+    public void updateConfig(Map<String, Object> props) {
+        final long maxMS = TimeUnit.NANOSECONDS.toMillis(Long.MAX_VALUE);
+
+        Object v;
+        int u_expedite = (Integer) props.get("expedite");
+        int u_max = null == (v = props.get("max")) ? Integer.MAX_VALUE : (Integer) v;
+        MaxPolicy u_maxPolicy = MaxPolicy.valueOf((String) props.get("maxPolicy"));
+        int u_maxQueueSize = null == (v = props.get("maxQueueSize")) ? Integer.MAX_VALUE : (Integer) v;
+        long u_maxWaitForEnqueue = (Long) props.get("maxWaitForEnqueue");
+        boolean u_runIfQueueFull = (Boolean) props.get("runIfQueueFull");
+        long u_startTimeout = null == (v = props.get("startTimeout")) ? -1l : (Long) v;
+
+        // Validation that cannot be performed by metatype:
+        if (u_expedite > u_max)
+            throw new IllegalArgumentException(u_expedite + " > " + u_max);
+
+        if (u_maxWaitForEnqueue < 0 || u_maxWaitForEnqueue > maxMS)
+            throw new IllegalArgumentException(Long.toString(u_maxWaitForEnqueue));
+
+        if (u_startTimeout < -1 || u_startTimeout > maxMS)
+            throw new IllegalArgumentException(Long.toString(u_startTimeout));
+
+        for (long current = maxWaitForEnqueueNS.get(); current != -1; current = maxWaitForEnqueueNS.get())
+            if (maxWaitForEnqueueNS.compareAndSet(current, TimeUnit.MILLISECONDS.toNanos(u_maxWaitForEnqueue)))
+                break;
+
+        maxPolicy = u_maxPolicy;
+        runIfQueueFull = u_runIfQueueFull;
+        startTimeout = TimeUnit.MILLISECONDS.toNanos(u_startTimeout);
+
+        int a;
+        synchronized (configLock) {
+            a = expeditesAvailable.addAndGet(u_expedite - expedite);
+            expedite = u_expedite;
+
+            int increase = u_max - maxConcurrency;
+            if (increase > 0)
+                maxConcurrencyConstraint.release(increase);
+            else if (increase < 0)
+                maxConcurrencyConstraint.reducePermits(-increase);
+            maxConcurrency = u_max;
+
+            increase = u_maxQueueSize - maxQueueSize;
+            if (increase > 0)
+                maxQueueSizeConstraint.release(increase);
+            else if (increase < 0)
+                maxQueueSizeConstraint.reducePermits(-increase);
+            maxQueueSize = u_max;
+        }
+
+        // Expedite as many of the remaining tasks as the available maxConcurrency permits and increased expedites
+        // will allow. We are choosing not to revoke GlobalPoolTasks that have already been enqueued as non-expedited,
+        // which means we do not guarantee an increase in expedites to fully go into effect immediately.
+        // Any reduction to expedites is handled gradually, as expedited GlobalPoolTasks complete.
+        while (withheldConcurrency.get() > 0 && maxConcurrencyConstraint.tryAcquire()) {
+            decrementWithheldConcurrency();
+            if (a-- > 0)
```

Comment:
Since this is outside of the config lock, would it be better to call acquireCoreCurrency() here instead of using the variable.  I'm not positive if this code could be accessed concurrently or not, so it may be ok, but wanted to be sure that it was considered.

Reply:
Good catch - the acquireExpedite() call was missed when copying from the config change logic from the expedite() method.